src/BoundaryConditions/BoundaryConditions.jl

```diff
@@ -0,0 +1,571 @@
+    bottom = regularize_bulk_bc(fbcs.bottom, grid, ρ₀, constants)
+    top = regularize_bulk_bc(fbcs.top, grid, ρ₀, constants)
+    # Keep other boundaries as-is
+    return FieldBoundaryConditions(west = fbcs.west,
+                                   east = fbcs.east,
+                                   south = fbcs.south,
+                                   north = fbcs.north,
+                                   bottom = bottom,
+                                   top = top,
+                                   immersed = fbcs.immersed)
+end
+
+# Pass through non-bulk-flux boundary conditions
+regularize_bulk_bc(bc, grid, ρ₀, constants) = bc
+
+# Regularize BulkSensibleHeatFlux
+function regularize_bulk_bc(bc::BulkSensibleHeatFluxBoundaryCondition, grid, ρ₀, constants)
+    bf = bc.condition.func
+    
+    # Convert surface_temperature to Field if it's a function
+    T₀ = materialize_surface_field(bf.surface_temperature, grid)
+    
+    # Use provided reference_density or compute from formulation
+    ref_density = isnothing(bf.reference_density) ? ρ₀ : bf.reference_density
+    
+    new_bf = BulkSensibleHeatFluxFunction(bf.coefficient, bf.gustiness, T₀, ref_density)
+    return BoundaryCondition(Flux(), DiscreteBoundaryFunction(new_bf, nothing))
+end
+
+# Regularize BulkVaporFlux
+function regularize_bulk_bc(bc::BulkVaporFluxBoundaryCondition, grid, ρ₀, constants)
+    bf = bc.condition.func
+    
+    # Convert surface_temperature to Field if it's a function
+    T₀ = materialize_surface_field(bf.surface_temperature, grid)
+    
+    # Use provided reference_density or compute from formulation
+    ref_density = isnothing(bf.reference_density) ? ρ₀ : bf.reference_density
+    
+    # Compute saturation specific humidity if not provided
+    if isnothing(bf.surface_specific_humidity)
+        qᵛ₀ = compute_saturation_specific_humidity(T₀, ref_density, constants, grid)
+    else
+        qᵛ₀ = materialize_surface_field(bf.surface_specific_humidity, grid)
+    end
+    
+    new_bf = BulkVaporFluxFunction(bf.coefficient, bf.gustiness, T₀, qᵛ₀, ref_density)
+    return BoundaryCondition(Flux(), DiscreteBoundaryFunction(new_bf, nothing))
+end
+
+# Helper to convert functions/numbers to Fields
+materialize_surface_field(f::Field, grid) = f
+materialize_surface_field(f::Number, grid) = f
+
+function materialize_surface_field(f::Function, grid)
+    field = Field{Center, Center, Nothing}(grid)
+    set!(field, f)
+    return field
+end
+
+function compute_saturation_specific_humidity(T₀::Field, ρ₀, constants, grid)
+    surface = PlanarLiquidSurface()
+    qᵛ₀ = Field{Center, Center, Nothing}(grid)
+    
+    # Create a function that computes qᵛ₀ pointwise
+    qᵛ₀_func(i, j) = saturation_specific_humidity(T₀[i, j, 1], ρ₀, constants, surface)
+    
+    # Set the field using the interior indices
+    for i in 1:size(grid, 1), j in 1:size(grid, 2)
+        qᵛ₀[i, j, 1] = qᵛ₀_func(i, j)
+    end
```

Comment:
GPU doesn't like the pointwise computation

Reply:
sorry i should be nice

Reply:
this will probably require a kernel